### PR TITLE
Add support for Github pull request comments

### PR DESCRIPTION
**Pull request by dancryer**
Not to be merged yet, initial support for PHPCS is done... Need to add support for: a) Other tools, b) Other platforms (Bitbucket, etc).

### CODE REVIEW

**Comment by dancryer** [comment]
FYI @mavimo for comparison with your Gitlab work.

**Comment by dancryer** [comment]
It seems it also needs work on detecting which diff line it should actually comment on. At the moment it assumes the first line that matches is the one, but for lines like closing braces `}`, etc. that falls down.

**Comment by dancryer** [comment]
See https://github.com/Block8/PHPCI/pull/803/files?diff=unified for an example.

File: PHPCI/Helper/Github.php [hunk 1]
```diff
@@ -108,4 +108,30 @@ public function getRepositories()
 
         return $rtn;
     }
+
+    public function createPullRequestComment($repo, $pullId, $commitId, $file, $line, $comment)
+    {
+        $token = Config::getInstance()->get('phpci.github.token');
```
Comment (mavimo) [inline PHPCI/Helper/Github.php]: This need to expose the github token in project, I think we can consider to check an ENV variable and if not set check for the config, maybe this improvement can be in b8framework (let it make *twelve factor app* compliant)

**Pull request closed without merging**